src/components/form.js

```diff
@@ -0,0 +1,51 @@
+import React, { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { v4 as uuidv4 } from 'uuid';
+import { addBookAction } from './redux/books/books';
+
+function Form() {
+  const dispatch = useDispatch();
+
+  const [book, setstate] = useState({ title: '', author: '' });
+
+  const changeTitle = (event) => setstate({ ...book, title: event.target.value });
+  const changeAuthor = (event) => setstate({ ...book, author: event.target.value });
+  const onChange = () => {
+    if (book.title && book.author) {
+      book.id = uuidv4();
+      dispatch(addBookAction(book));
+      setstate({ title: '', author: '' });
+    }
+  };
+
+  // const onchange = (e) => {
+  //   setstate((state) => ({
+  //     ...state,
+  //     [e.target.name]: e.target.value,
+  //     id: uuidv4(),
+  //   }));
+  // };
+
```

Comment:
- [ ] Kindly remove any commented ((unused) code to make your code more readable and clean.